staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery_test.go

```diff
@@ -151,17 +152,18 @@ func TestDirty(t *testing.T) {
 			},
 		},
 	}, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		pinged = true
+		pinged.Store(true)
 		service.ServeHTTP(w, r)
 	}))
 	testCtx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	go aggregatedManager.Run(testCtx.Done())
 	require.True(t, waitForEmptyQueue(testCtx.Done(), aggregatedManager))
+	aggregatedManager.dirtyAPIServiceQueue.ShutDownWithDrain()
```

Comment:
While this blocks until all items in the queue have synced, I don't think it is the right general solution. `waitForEmptyQueue` needs to be modified to wait for queue to stop processing, otherwise the other tests that use `waitForEmptyQueue` will still be affected

Reply:
I think so too. I am trying to solve this with more general way. See https://github.com/kubernetes/kubernetes/pull/115859#issuecomment-1434881423